### PR TITLE
Create a fake btcd.

### PR DESCRIPTION
**Pull request by fox000002**
Issue #86

### CODE REVIEW

**Comment by gz-c** [comment]
What is the command to add a fake deposit?

**Comment by fox000002** [comment]
It's command "nextdeposit". @gz-c

**Comment by gz-c** [comment]
What are the arguments/how to call it

**Comment by fox000002** [comment]
It's called by 

`btcrpc.RawRequest("nextdeposit", nil)`

**Comment by gz-c** [comment]
How do i specify the address and amount?

**Comment by fox000002** [comment]
That's not supported yet. It just trigger a new block from predefined data currently.

**Comment by gz-c** [comment]
Oh, can you let it control the address, amount and vout index?
The rest of the data in the block can be fake.  It only needs enough data for the scanner to understand it.

**Comment by gz-c** [comment]
Maybe we can wrap btcrpcclient into an interface, and substitute a fake one if we set some config flag. And the fake one has an http api to add a fake deposit

**Comment by fox000002** [comment]
OK. I will do that.

**Comment by fox000002** [comment]
Where to add the fake btcrpcclient interface?

**Comment by gz-c** [comment]
Refactor the btcrpcclient into a new package

**Comment by gz-c** [comment]
Add a new package

File: cmd/btcd/btcd.go [hunk 1]
```diff
@@ -1269,29 +1182,140 @@ func (c *wsClient) WaitForShutdown() {
...
+	if err != nil {
+		return err
+	}
+
+	_, err = w.Write(d)
+	return err
+}
+
+func httpHandleNextDeposit(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Connection", "close")
```
Comment (gz-c) [inline cmd/btcd/btcd.go]: Check `r.Method == http.MethodPost`

**Pull request merged**